include/Code.h

```diff
@@ -0,0 +1,15 @@
+#ifndef CODE_H
+#define CODE_H
+
+enum class Code {
+	CODE_ZERO = 0,
```

Comment:
Using an enum for the return codes is great, but it defeats the purpose if you don't give the enums meaningful names. 
For example, the code improve in readability if instead of `CODE_42` you named it `LegalMove`